RELEASE/scripts/autoscend/iotms/mr2021.ash

```diff
@@ -112,3 +112,245 @@ int auto_backupUsesLeft()
 {
 	return 11 + (my_path() == "You, Robot" ? 5 : 0) - get_property("_backUpUses").to_int();
 }
+
+boolean auto_havePowerPlant()
+{
+	return possessEquipment($item[potted power plant]) && auto_is_valid($item[potted power plant]);
+}
+
+void auto_harvestBatteries()
+{
+	if (auto_havePowerPlant())
+	{
+		// Stolen straight from mafia's breakfast handling.
+		cli_execute("inv_use.php?pwd&whichitem=" + $item[potted power plant].to_int());
+		
+		string [int] status = split_string(get_property("_pottedPowerPlant"), ",");
+
+		for ( int pp = 0; pp < status.count(); pp++ )
+		{
+			if ( status[pp] > 0)
+			{
+				cli_execute("choice.php?pwd&whichchoice=1448&option=1&pp=" + ( pp + 1 ));
+			}
+		}
+	}
+}
+
+// These points the value of a battery represented in AAAs.
+int batteryPoints(item battery)
+{
+	int points = 0;
+	switch (battery)
+	{
+		case $item[battery (AAA)]:
+			points = 1;
+			break;
+		case $item[battery (AA)]:
+			points = 2;
+			break;
+		case $item[battery (D)]:
+			points = 3;
+			break;
+		case $item[battery (9-Volt)]:
+			points = 4;
+			break;
+		case $item[battery (lantern)]:
+			points = 5;
+			break;
+		case $item[battery (car)]:
+			points = 6;
+			break;
+	}
+	return points;

```

Comment:
This can be replaced with a static map. See `ed_KaCost` or `ed_shopping` for examples

Reply:
That's a lot cleaner, I think I made the conversion correctly, but maps are a still a bit of a strange concept to me.

Reply:
resolved